docs/design/auto-storage-scale.md

```diff
@@ -0,0 +1,49 @@
+# Automatic Storage Scaling
+
+# Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+# Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+# High-Level Design
+
+- Create a AutoStorageScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: AutoStorageScaling
+    Metadata:
+	    Name: <storageclusterName>-<deviceClassName>
+	    namespace: <namespace>
+    Spec: 
+	    storageClusterName: <storageclusterName>
```

Comment:
Only one storage cluster CR is supported, right? If so, we don't need the name

Reply:
@travisn From my previous conversations with @parth-gr, `AutoStorageScale` CR will be associated to a `StorageCluster` and a deviceclass, so it's required as the UI needs to retrieve the correct CR looking filtering by storage cluster name and deviceclass.

Reply:
Yaa lets add the support, even not needed right now, to make it generic

Reply:
@alfonsomthd Sounds good to add the storageClusterName just to be clear, but does ODF even support multiple storageCluster CRs? I can't think where that would be needed, except perhaps when there is both an external and internal cluster, but even those are in different namespaces.

Reply:
https://github.com/red-hat-storage/ocs-operator/pull/3044?new_mergebox=false#discussion_r1967449209